### PR TITLE
[workflow] expand workflow options

### PR DESCRIPTION
**Pull request by dcramer**
This adds the ability to specify the release an issue is resolved in. Right now it requires that the release exists in the system.

It also makes statusDetails a mapping that you provide for status changes, vs adding them all to top level. Historic usage is still supported.

```
PUT ...
{
  "status": resolved,
  "statusDetails": {
    "inRelease": "latest"
  }
}
```

```
PUT ...
{
  "status": resolved,
  "statusDetails": {
    "inNextRelease": true
  }
}
```

Under the hood the above do the same thing except they store a slightly different Activity entry.

It also improves visuals around the ignore action.

### CODE REVIEW

**Comment by dcramer** [comment]
Still need UI

**Comment by dcramer** [comment]
screenshot 2017-06-01 15 17 02

screenshot 2017-06-01 15 17 08

**Comment by dcramer** [comment]
I cant figure out whats wrong with the typehead box. It wont let you enter any actual text in this case.

**Comment by dcramer** [comment]
Expanding what this branch does to try and unify how ignore/resolution are represented

@ckj

**Comment by benvinegar** [comment]
Could we render a short form of the current release string so you get something like:

> The current release (abc123&hellip;)

**Comment by dcramer** [comment]
@benvinegar i like that idea but it turns out we dont know that information at the current scope

**Comment by dcramer** [comment]
I've fixed all the quirks. tabindex=-1 on modal was causing select2 to be uneditable. Lack of "id" for the custom rendering bits was causing things to be unselectable.

File: src/sentry/static/sentry/app/components/customResolutionModal.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    projectId: React.PropTypes.string.isRequired,
+    show: React.PropTypes.bool
+  },
+
+  getInitialState() {
+    return {version: ''};
+  },
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.show) {
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/customResolutionModal.jsx]: Do you not want to see if there was a change between `prevProps` and `this.props`? Cause `componentDidUpdate` can trigger repeatedly in succession where `this.props.show` doesn't actually change (literally any prop/state change will trigger it).

(Not that this is doing anything heavy.)

File: src/sentry/static/sentry/app/components/customResolutionModal.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      // XXX(cramer): this is incorrect but idgaf
+      jQuery('.modal').attr('tabindex', null);
+    }
+  },
+
+  onSubmit() {
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/customResolutionModal.jsx]: Aside/musing/ignore this – in the future, we should having a naming policy that distinguishes between a prop name and a prop handler.

Something like `onX` for prop names/event names, `handleX` for the prop/event handler:

```javascript
<button onCancel={this.handleCancel}/>
```

**Review by benvinegar — APPROVED** [review]
Only reviewed front-end stuff – see comments.

Also, can we not get a snapshot test for this?

File: src/sentry/static/sentry/app/components/customResolutionModal.jsx (hunk 1 above)
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/customResolutionModal.jsx]: Actually, going back to this, yes, you should make this change. Modifying the DOM (attr change) repeatedly/unnecessarily is not ideal.

Comment (dcramer) [inline src/sentry/static/sentry/app/components/customResolutionModal.jsx]: done

File: src/sentry/static/sentry/app/components/customResolutionModal.jsx (hunk 2 above)
Comment (dcramer) [inline src/sentry/static/sentry/app/components/customResolutionModal.jsx]: agreed, we should spec it and then follow it

Comment (benvinegar) [inline src/sentry/static/sentry/app/components/customResolutionModal.jsx]: 👍

**Pull request merged**